controller/backing_image_controller.go

```diff
@@ -310,22 +345,530 @@ func (bic *BackingImageController) syncBackingImage(key string) (err error) {
+		return errors.Wrapf(err, "failed to delete invalid v2 backing image")
+	}
+
+	if err := bic.prepareV2Copy(bi); err != nil {
+		return errors.Wrapf(err, "failed to prepare v2 backing image")
+	}
+
+	return nil
+}
+
+func (bic *BackingImageController) isFirstV2CopyInState(bi *longhorn.BackingImage, state longhorn.BackingImageState) bool {
+	if bi.Status.V2FirstCopyDisk != "" && bi.Status.V2FirstCopyStatus == state {
+		return true
+	}
+
+	if bi.Status.V2FirstCopyDisk != "" {
+		if status, exists := bi.Status.DiskFileStatusMap[bi.Status.V2FirstCopyDisk]; exists && status.State == state {
+			return true
+		}
+	}
+
+	return false
+}
+
+func (bic *BackingImageController) cleanupFirstFailedV2Copy(bi *longhorn.BackingImage) (err error) {
+	if bi.Status.V2FirstCopyDisk == "" {
+		return nil
+	}
+
+	log := getLoggerForBackingImage(bic.logger, bi)
+
+	node, _, err := bic.ds.GetReadyDiskNodeRO(bi.Status.V2FirstCopyDisk)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get the ready disk node for disk %v", bi.Status.V2FirstCopyDisk)
+	}
+
+	engineClientProxy, err := bic.getEngineClientProxy(node.Name, longhorn.DataEngineTypeV2, log)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get the engine client proxy for node %v", node.Name)
+	}
+	defer engineClientProxy.Close()
+
+	// clean up the failed first v2 copy
+	err = engineClientProxy.SPDKBackingImageDelete(bi.Name, bi.Status.V2FirstCopyDisk)
+	if err != nil {
+		return errors.Wrapf(err, "failed to delete the failed v2 copy on the disk %v", bi.Status.V2FirstCopyDisk)
+	}
+	delete(bi.Spec.DiskFileSpecMap, bi.Status.V2FirstCopyDisk)
+	delete(bi.Status.DiskFileStatusMap, bi.Status.V2FirstCopyDisk)
+	bi.Status.V2FirstCopyDisk = ""
+	bi.Status.V2FirstCopyStatus = longhorn.BackingImageStatePending
+	return nil
+}
+
+func (bic *BackingImageController) deleteAllV1FileCopies(bi *longhorn.BackingImage) {
+	for diskUUID, fileSpec := range bi.Spec.DiskFileSpecMap {
+		if types.IsDataEngineV1(fileSpec.DataEngine) {
+			delete(bi.Spec.DiskFileSpecMap, diskUUID)
+		}
+	}
+}
+
+func (bic *BackingImageController) prepareFirstV2Copy(bi *longhorn.BackingImage) (err error) {
+	log := getLoggerForBackingImage(bic.logger, bi)
+
+	// The preparation state transition: Pending -> InProgress -> Ready/Failed
+	// we retry when failed by deleting the failed copy and cleanup the state.
+
+	// If the first v2 copy is ready, we can delete all the v1 file copies and return.
+	isPrepared := bic.isFirstV2CopyInState(bi, longhorn.BackingImageStateReady)
+	if isPrepared {
+		bi.Status.V2FirstCopyStatus = longhorn.BackingImageStateReady
+		bic.deleteAllV1FileCopies(bi)
+		bic.v2CopyBackoff.DeleteEntry(bi.Status.V2FirstCopyDisk)
+		return nil
+	}
+
+	// If the first v2 copy is failed, we cleanup the failed copy and retry the preparation.
+	// If the first v2 copy is in unknown, that means the instance manager may crahsh when creating the first v2 copy.
+	// We cleanup the unknown copy and retry the preparation.
```

Comment:
I have written down it in another comment , in `deleteInvalidV2Copy`
```
	// When the backing image is created on that disk, it is either ready, failed or unknown.
	// We directly delete the failed v2 copy if there is ready copy
	// For unknown v2 copy, it can happen in the following case,
	// 1. If the instance-manager pod is not running or the instance-manager CR is not ready.
	// 2. Backing image copy has status (ready or failed) before but can't get the ststus now.
	// 3. Backing image copy is failed to be created on the disk and there is no lvol snapshot and at the same time the node is rebooted.
	//    It will keeps in unknown state forever because spdk server has no snapshot lvol to pick up to reconstruct the record.
	// For case 1 and case 2, the status will come back automatically when server pick the snapshot lvol up and reconstruct the record.
	// We delete the unknown copy only for the case 3 if there is a ready copy so it can be recreated.
```

Reply:
you can check the `syncV2StatusWithInstanceManager()`